workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.test.ts

```diff
@@ -0,0 +1,124 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "../Iterator.prototype.filter";
+import "../Iterator.prototype.map";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).toSet;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.toSet", () => {
+  it("adds primitives to a Set", () => {
+    const arr = [1, 2, 2, 3, 1, 2, 3, 3, 100, false, 100, false, "hello"];
+    const iteratorToSet = arr.values().toSet();
+
+    expect(iteratorToSet).toBeInstanceOf(Set);
+    expect(iteratorToSet).toStrictEqual(
+      new Set([1, 2, 3, 100, false, "hello"]),
+    );
+  });
+
+  it("adds objects to a Set", () => {
+    const obj1 = { a: 1 };
+    const obj2 = { b: 2 };
+    const objectArray = [obj1, obj2, obj1, obj2, obj2, obj1];
+    const iteratorToSet = objectArray.values().toSet();
+
+    expect(iteratorToSet).toBeInstanceOf(Set);
+    expect(iteratorToSet).toStrictEqual(new Set([obj1, obj2]));
+  });
+
+  it("adds arrays to a Set", () => {
+    const arr1 = [1, 2];
+    const arr2 = [3, 4];
+    const arrayOfArrays = [
+      arr1,
+      arr2,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr2,
+      arr2,
+      arr2,
+    ];
+    const iteratorToSet = arrayOfArrays.values().toSet();
+
+    expect(iteratorToSet).toBeInstanceOf(Set);
+    expect(iteratorToSet).toStrictEqual(new Set([arr1, arr2]));
+  });
+
+  it("can convert multiple Generators to a Set", () => {
+    const generator1 = function* (): Generator<string, void, void> {
+      yield "a";
+      yield "b";
+      yield "c";
+      yield "a";
+    };
+
+    const generator2 = function* (): Generator<string, void, void> {
+      yield "x";
+      yield "y";
+      yield "z";
+      yield "x";
+    };
+
+    const combinedIterator = function* (): Generator<string, void, void> {
```

Comment:
As an aside, being able to concatenate two iterators would make for a fun goody!

Reply:
should we make an issue for that?

Reply:
Sure, you're welcome to create it if you'd like, or let me know if you want me to. I'd probably call it `concat`, since [there's precedent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat).